frontends/ol-search-ui/src/components/ShareTooltip.tsx

```diff
@@ -0,0 +1,87 @@
+import React from "react"
+import {
+  FacebookShareButton,
+  LinkedinShareButton,
+  TwitterShareButton
+} from "react-share"
+import { FacebookIcon, TwitterIcon, LinkedinIcon } from "react-share"
+import Tooltip from "rc-tooltip"
+
+type HelperProps = {
+  url: string
+  hideSocialButtons?: boolean | null | undefined
+  objectType?: string | null | undefined
+}
+
+export class ShareTooltipHelper extends React.Component<HelperProps> {
+  input: { current: null | React.ReactNode }
+
+  constructor(props: HelperProps) {
+    super(props)
+    this.input = React.createRef()
+  }
+
+  selectAndCopyLinkText = (e: React.MouseEvent<HTMLElement>) => {
+    e.preventDefault()
+    if (this.input.current) {
+      // @ts-expect-error not sure why typscript expects a type here
+      this.input.current.select()
```

Comment:
The reason you needed `ts-expect-error` is that the ref has type HTMLElement, but not all HTMLElments have a `select` method.

The ref is actually an `HTMLInputElement`, which does have select.

Also:
**Suggestion:** Try `navigator.clipboard.writeText(this.input.current.value)` (see [writeText](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/writeText)). 

[execCommand is deprecated](https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand)